toolkit/scripts/containerized-build/create_container_build.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -e
+
+script_dir=$(realpath "$(dirname "$0")")
+topdir=/usr/src/mariner
+enable_local_repo=false
+
+switch_to_red_text() {
+    printf "\e[31m"
+}
+
+switch_to_normal_text() {
+    printf "\e[0m"
+}
+
+print_error() {
+    echo ""
+    switch_to_red_text
+    echo ">>>> ERROR: $1"
+    switch_to_normal_text
+    echo ""
+}
+
+help() {
+echo "
+Usage:
+sudo make containerized-rpmbuild [REPO_PATH=/path/to/CBL-Mariner] [MODE=test|build] [VERSION=1.0|2.0] [MOUNTS= /src/path:/dst/path] [ENABLE_REPO=y]
+
+Starts a docker container with the specified version of mariner.
+
+Optional arguments:
+    REPO_PATH:      path to the CBL-Mariner repo root directory. default: "current directory"
+    MODE            build or test. default:"build"
+                        In 'test' mode it will use a pre-built mariner chroot image.
+                        In 'build' mode it will use the latest published container.
+    VERISION        1.0 or 2.0. default: "2.0"
+    MOUNTS          mount a directory into the container. Should be of the form '/src/dir:/dest/dir'. 
+                    Can be specified multiple times.
+    ENABLE_REPO:    Set to 'y' to use local RPMs to satisfy package dependencies. default: "n"
+
+To see help, run 'sudo make containerized-rpmbuild-help'
+"
+}
+
+build_chroot() {
+    pushd "${repo_path}/toolkit"
+    echo "Building worker chroot..."
+    sudo make graph-cache REBUILD_TOOLS=y > /dev/null
+    popd
+}
+
+build_tools() {
+    pushd "${repo_path}/toolkit"
+    echo "Building tools..."
+    sudo make go-depsearch REBUILD_TOOLS=y > /dev/null
+    sudo make go-grapher REBUILD_TOOLS=y > /dev/null
+    sudo make go-specreader REBUILD_TOOLS=y > /dev/null
+    popd
+}
+
+build_graph() {
+    pushd "${repo_path}/toolkit"
+    echo "Building dependency graph..."
+    sudo make workplan > /dev/null
+    popd
+}
+
+while (( "$#")); do
+  case "$1" in
+    -m ) mode="$2"; shift 2 ;;
+    -v ) version="$2"; shift 2 ;;
+    -p ) repo_path="$(realpath $2)"; shift 2 ;;
+    --enable-local-repo ) enable_local_repo=true; shift ;;
+    --help ) help; exit 1 ;;
+    ? ) echo -e "ERROR: INVALID OPTION.\n\n"; help; exit 1 ;;
+  esac
+done
+
+# Assign default values
+[[ -z "${repo_path}" ]] && repo_path=$(realpath "$(dirname "$0")") && repo_path=${repo_path%'/toolkit'*}
```

Comment:
Should you use `${script_dir}` here instead of recalculating the value? Also, don't you need to backtrack a couple of directories from the script directory to get to the toolkit directory?

Reply:
${script_dir} is different from ${repo_path}, because we might want to use another Mariner repo to build RPMs for.
Yes, that's what repo_path=${repo_path%'/toolkit'*} achieves

Reply:
I meant do `repo_path=${script_dir}`